tests/integration/test.replication.js

```diff
@@ -2519,6 +2519,326 @@ adapters.forEach(function (adapters) {
       });
     });
 
+    // Should not start replication over if last_seq mismatches in checkpoints
+    // and it can be resolved some other way
+    it('#3999-1 should not start over if last_seq mismatches', function () {
+
+      var source = new PouchDB(dbs.remote);
+      var mismatch = false;
+      var failWrite = false;
+      var checkpoint;
+
+      // 1. This is where we fake the mismatch:
+      var putte = source.put;
+      source.put = function (doc) {
+
+        // We need the checkpoint id so we can inspect it later
+        if (/local/.test(doc._id)) {
+          checkpoint = doc._id;
+        }
+
+        if (!failWrite || doc.last_seq !== 2) {
+          return putte.apply(this, arguments);
+        }
+
+        return PouchDB.utils.Promise.reject({
+          status: 0,
+          message: 'Database encountered an unknown error'
+        });
+      };
+
+      // 2. We measure that the replication starts in the expected
+      // place in the 'changes' function
+      var changes = source.changes;
+      source.changes = function (opts) {
+
+        if (mismatch) {
+          opts.since.should.be.at.least(1);
```

Comment:
We cant use numerical comparisons for couchdb-master checkpoints since they are arrays / objects, we can use `collate` to compare them, although in this case I think 

    opts.since.should.not.equal(0);

is fine